packages/next/server/web/spec-compliant/request.ts

```diff
@@ -16,7 +16,9 @@ class BaseRequest extends Body implements Request {
   }
 
   constructor(input: BaseRequest | string, init: RequestInit = {}) {
-    const method = init.method?.toUpperCase() ?? 'GET'
+    const method =
+      init.method?.toUpperCase() ??
+      (input instanceof BaseRequest ? input.method?.toUpperCase() : 'GET')
```

Comment:
```suggestion
    const method = init.method?.toUpperCase() ?? 
    (typeof input !== 'string' ? input.method?.toUpperCase() : 'GET')
```

Reply:
Would we also need to update the checks to consider `input` being nullish if we check for `!== 'string'` here??